bin/query-blocks.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Queries for the blocks on the site.
+ *
+ * @package Genesis\CustomBlocks
+ */
+
+namespace Genesis\CustomBlocks\Cli;
+
+use WP_CLI;
+use WP_Query;
+
+/**
+ * Gets the field types (controls) and their counts.
+ *
+ * @return array[] $fields {
+ *     The fields in all of the blocks.
+
+ *     @type string $field The type (control) of the field, like 'text'.
+ *     @type int    $count The count of that type in all of the blocks.
+ * }
+ */
+function get_fields() {
+	$block_query = new WP_Query(
+		[
+			'post_type'      => 'genesis_custom_block',
+			'post_status'    => 'publish',
+			'posts_per_page' => 99,
+		]
+	);
+
+	$field_histogram = [];
+	foreach ( $block_query->posts as $post ) {
+		$block_json = json_decode( $post->post_content, true );
+		if ( ! isset( $block_json[ 'genesis-custom-blocks/' . $post->post_name ]['fields'] ) ) {
+			break;
+		}
+
+		foreach ( $block_json[ 'genesis-custom-blocks/' . $post->post_name ]['fields'] as $field_name => $field ) {
+			$field_histogram[ $field['control'] ] = ! empty( $field_histogram[ $field['control'] ] ) && is_int( $field_histogram[ $field['control'] ] )
+				? $field_histogram[ $field['control'] ] + 1
+				: 1;
+		}
+	}
+
+	$table_fields = [];
+	foreach ( $field_histogram as $field => $count ) {
+		$table_fields[] = compact( 'field', 'count' );
+	}
+
+	return $table_fields;
+}
+
+/**
+ * Queries the blocks on the site and logs the results.
+ */
+function query_blocks() {
+	$post_count = wp_count_posts( 'genesis_custom_block' );
+	if ( empty( $post_count->publish ) ) {
+		WP_CLI::log( 'There is no GCB block' );
+		return;
+	}
+
+	WP_CLI::log( sprintf( 'There are %1$d GCB blocks', $post_count->publish ) );
+	WP_CLI::log( "\n" );
+
+	$table_values = get_fields();
+	$table_fields = [ 'field', 'count' ];
+	$assoc_args   = [ 'fields' => implode( ',', $table_fields ) ];
+
+	WP_CLI::log( 'Total counts of each field type (control):' );
+	$formatter = new WP_CLI\Formatter( $assoc_args, $table_fields );
+	$formatter->display_items( $table_values );
+}
+
+if ( ! defined( 'WP_CLI' ) ) {
+	echo "Please run this with WP-CLI: wp eval-file bin/query-blocks.php\n";
+	exit( 1 );
+}
+
+if (
+	! is_plugin_active( 'genesis-custom-blocks/genesis-custom-blocks.php' )
+	&& ! is_plugin_active( 'genesis-custom-blocks-pro/genesis-custom-blocks-pro.php' )
```

Comment:
If either is active 😄

Reply:
Would it then need to be:

```
! if_is_plugin_active() || ! is_plugin_active()
```
Instead of `&&`

Reply:
Ah, that's only a notice that displays if neither GCB free or Pro is active. 

The script still runs if they're inactive, it's just that it won't find any blocks because `register_block_type( 'genesis_custom_block' )` won't run.

I'm deleting it, as it was confusing. The real home of this script will probably be in `php/`, which will require the plugin to be active.